gradle/libs.versions.toml

```diff
@@ -42,12 +42,13 @@ logback = "1.4.11"
 maven = "3.9.4"
 mavenResolver = "1.9.15"
 mockk = "1.13.7"
-okhttp = "4.11.0"
+okhttp = "5.0.0-alpha.11"
```

Comment:
Why's this necessary? We shouldn't use alpha version dependencies in production code (and ORT's `main` *is* used in production, as it's supposed to always be stable).

Reply:
Amazon has [stated](https://github.com/awslabs/aws-sdk-kotlin/issues/765#issuecomment-1374093175) that their Koltin library relies on the latest version of okhttp, which oddly enough is the alpha one.  I'm not sure how to proceed.

Reply:
> which oddly enough is the alpha one.

Thanks for this information, indeed odd. I don't think it's acceptable for us to depend on an alpha version of OkHttp for the whole of ORT. We could solve this if 1) [scan storages were plugin](https://github.com/oss-review-toolkit/ort/issues/6603) and 2) [plugin classpaths were isolated from ORT classpath](https://github.com/oss-review-toolkit/ort/issues/6798), but both are bigger refactorings that I do not see happening any time soon.

> I'm not sure how to proceed.

I guess our best option then is to indeed use the Java instead of the Kotlin flavor of the AWS SDK...

Reply:
I'll start working on a refactor.

Reply:
I have completed the refactor, please review.

Reply:
I'm sorry, I'm afraid I need some clarification. There are two commits on the branch right now, the first one contains the version that uses the experimental Kotlin library by Amazon. Then another commit with a refactor switching to Java library, therefore, this alpha version (and other changes mentioned below) are found **solely** in the first commit but **not** in the HEAD. Should this first commit be squashed too? What I usually do is to keep commits until just before merging, then I squash them, just in case I need to track down errors and even discarded ideas. Could you please explain me this project's policy regarding that?

Reply:
> There are two commits on the branch right now, the first one contains the version that uses the experimental Kotlin library by Amazon.

Ah, my bad, I might have accidentally looked only at the first commit, assuming there would be only one / believing to look at the combined diff.

> Should this first commit be squashed too?

Yes please. We don't squash upon merge (as that combines individual commits from a PR), but rebase upon merge (also merge commit are not allowed).

> What I usually do is to keep commits until just before merging, then I squash them, just in case I need to track down errors and even discarded ideas.

Makes sense, but locally you could probably use `git reflog` to restore intermediate states.

Reply:
Got it! Thans for the `git reflog ` tip, life saver.